addons/binding/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/handler/TeslaHandler.java

```diff
@@ -540,28 +542,58 @@ public void parseAndUpdate(String request, String payLoad, String result) {
-                                            .getState(entry.getValue().getAsString(), selector, editProperties()));
-                                } else {
-                                    updateState(selector.getChannelID(), UnDefType.UNDEF);
-                                }
-                            } else {
-                                if (!entry.getValue().isJsonNull()) {
-                                    Map<String, String> properties = editProperties();
-                                    properties.put(selector.getChannelID(), entry.getValue().getAsString());
-                                    updateProperties(properties);
+                        if (entry.getKey().equals("timestamp")) {
+                            resultTimeStamp = Long.valueOf(entry.getValue().getAsString());
+                            Date date = new Date();
+                            date.setTime(resultTimeStamp);
+                            logger.trace("The request result timestamp is {}", DATE_FORMATTER.format(date));
+                            break;
+                        }
+                    }
+
+                    try {
+                        lock.lock();
+
+                        boolean proceed = true;
+                        if (resultTimeStamp < lastTimeStamp && request == TESLA_DRIVE_STATE) {
+                            proceed = false;
+                        } else {
+                            lastTimeStamp = resultTimeStamp;
+                        }
+
+                        if (proceed) {
+                            for (Map.Entry<String, JsonElement> entry : entrySet) {
+                                try {
+                                    TeslaChannelSelector selector = TeslaChannelSelector
+                                            .getValueSelectorFromRESTID(entry.getKey());
+                                    if (!selector.isProperty()) {
+                                        if (!entry.getValue().isJsonNull()) {
+                                            updateState(selector.getChannelID(), teslaChannelSelectorProxy.getState(
+                                                    entry.getValue().getAsString(), selector, editProperties()));
+                                        } else {
+                                            updateState(selector.getChannelID(), UnDefType.UNDEF);
+                                        }
+                                    } else {
+                                        if (!entry.getValue().isJsonNull()) {
+                                            Map<String, String> properties = editProperties();
+                                            properties.put(selector.getChannelID(), entry.getValue().getAsString());
+                                            updateProperties(properties);
+                                        }
+                                    }
+                                } catch (Exception e) {
```

Comment:
You catch the class Exception, do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception